src/Native/Unix/System.Security.Cryptography.Native/openssl.c

```diff
@@ -1332,8 +1331,8 @@ extern "C" int32_t CryptoNative_EnsureOpenSslInitialized()
     }
 
     // Create the locks array
-    g_locks = new (std::nothrow) pthread_mutex_t[numLocks];
-    if (g_locks == nullptr)
+    g_locks = (pthread_mutex_t*)malloc(sizeof(pthread_mutex_t) * (size_t)numLocks);
```

Comment:
Is there any situation where we'd need to worry about overflows here?

Reply:
That's not what I was talking about. Multiplication can overflow back to being a positive number, at which point the malloc would succeed but for less space than was actually needed, which can then result in buffer overruns. That's very unlikely to happen here, but I'd still feel better if we handled it. We have a helper for doing safe multiplication.

Reply:
Thanks for the explanation @stephentoub. I have pushed a commit using multiply_s.